openfeature/hooks/logging_hook.go

```diff
@@ -0,0 +1,97 @@
+package hooks
+
+import (
+	"context"
+	"log/slog"
+
+	of "github.com/open-feature/go-sdk/openfeature"
+)
+
+const (
+	DOMAIN_KEY             = "domain"
+	PROVIDER_NAME_KEY      = "provider_name"
+	FLAG_KEY_KEY           = "flag_key"
+	DEFAULT_VALUE_KEY      = "default_value"
+	EVALUATION_CONTEXT_KEY = "evaluation_context"
+	ERROR_CODE_KEY         = "error_code"
+	ERROR_MESSAGE_KEY      = "error_message"
+	REASON_KEY             = "reason"
+	VARIANT_KEY            = "variant"
+	VALUE_KEY              = "value"
+)
+
+type LoggingHook struct {
+	includeEvaluationContext bool
+	logger                   *slog.Logger
+}
+
+func NewLoggingHook(includeEvaluationContext bool) (*LoggingHook, error) {
+	return NewCustomLoggingHook(slog.Default(), includeEvaluationContext)
+}
+
+func NewCustomLoggingHook(logger *slog.Logger, includeEvaluationContext bool) (*LoggingHook, error) {
```

Comment:
One last nit: can you reverse the param order so `includeEvaluationContext bool` is always first? :sweat_smile:

Reply:
done